alerter.c

```diff
@@ -1,19 +1,26 @@
 #include <stdio.h>
 #include <assert.h>
-
+#define SW_TEST 0

```

Comment:
Using conditional compilation to execute test or production code increases complexity as the code size grows. Maintenance becomes difficult.
Test code and production code should be maintained separately.

Production code should not be aware of test environment or production envt.

Use dependency injection to solve this problem.